libnmstate/nm/veth.py

```diff
@@ -48,6 +49,13 @@ def create_setting(iface, base_con_profile):
 
     if not veth_setting:
         veth_setting = getattr(NM, SETTING_VETH).new()
+
+    if not veth_setting.props.peer and not iface.peer:
```

Comment:
This kind of check should done at nmstate level instead of nm level.

Reply:
Done, the checking is being done at ifaces.